packages/salesforcedx-vscode-core/src/commands/deployManifest.ts

```diff
@@ -39,18 +36,14 @@ class LibraryDeployManifestExecutor extends DeployExecutor<string> {
 }
 
 export const deployManifest = async (manifestUri: URI) => {
-  if (!manifestUri) {
-    const editor = vscode.window.activeTextEditor;
-    if (editor && editor.document.languageId === 'forcesourcemanifest') {
-      manifestUri = editor.document.uri;
-    } else {
-      const errorMessage = nls.localize('deploy_select_manifest');
-      telemetryService.sendException('deploy_with_manifest', errorMessage);
-      void notificationService.showErrorMessage(errorMessage);
-      channelService.appendLine(errorMessage);
-      channelService.showChannelOutput();
-      return;
-    }
+  const resolved =
+    manifestUri ??
+    (await getUriFromActiveEditor({
```

Comment:
there's a bunch like this, so I made a shared fn for them.  One of the very few human refactorings in this PR.